applications/ParticleMechanicsApplication/custom_elements/updated_lagrangian.hpp

```diff
@@ -81,6 +83,79 @@ class UpdatedLagrangian
     KRATOS_DEFINE_LOCAL_FLAG( COMPUTE_RHS_VECTOR_WITH_COMPONENTS );
     KRATOS_DEFINE_LOCAL_FLAG( COMPUTE_LHS_MATRIX_WITH_COMPONENTS );
 
+    struct MaterialPointVariables
+    {
+    public:
+        // Particle Position
+        CoordinatesArrayType xg;
+        // MP_MASS
+        double mass;
+        // MP_DENSITY
+        double density;
+        // MP_VOLUME
+        double volume;
+
+        // MP_DISPLACEMENT
+        array_1d<double, 3> displacement;
+        // MP_VELOCITY
+        array_1d<double, 3> velocity;
+        // MP_ACCELERATION
+        array_1d<double, 3> acceleration;
+
+        // MP_VOLUME_ACCELERATION
+        array_1d<double, 3> volume_acceleration;
+
+        // MP_CAUCHY_STRESS_VECTOR
+        Vector cauchy_stress_vector;
+        // MP_ALMANSI_STRAIN_VECTOR
+        Vector almansi_strain_vector;
+
+        // MP_DELTA_PLASTIC_STRAIN
+        double delta_plastic_strain;
+        // MP_DELTA_PLASTIC_VOLUMETRIC_STRAIN
+        double delta_plastic_volumetric_strain;
+        // MP_DELTA_PLASTIC_DEVIATORIC_STRAIN
+        double delta_plastic_deviatoric_strain;
+        // MP_EQUIVALENT_PLASTIC_STRAIN
+        double equivalent_plastic_strain;
+        // MP_ACCUMULATED_PLASTIC_VOLUMETRIC_STRAIN
+        double accumulated_plastic_volumetric_strain;
+        // MP_ACCUMULATED_PLASTIC_DEVIATORIC_STRAIN
+        double accumulated_plastic_deviatoric_strain;
+
+        explicit MaterialPointVariables(SizeType WorkingSpaceDimension)
+        {
+            // MP_MASS
+            mass = 1.0;
+            // MP_DENSITY
+            density = 1.0;
+            // MP_VOLUME
+            volume = 1.0;
+
+            SizeType strain_size = (WorkingSpaceDimension == 2)
```

Comment:
This will limit you to access the third direction strain and stress under plane stress and plane strain assumption. For certain material models like CamClay, you might need the third directions component even in 2D.

Reply:
Thanks for the heads up. From your experience, how was this handled before? The element must have known the correct strain size somehow?

Reply:
I checked it, it should be fine.